core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java

```diff
@@ -139,24 +148,30 @@ private void collectAndRemoveOldVersionData() throws IOException {
     long transactionIdToDelete = deleteTimeLine << 18;
     LOG.info("Start to remove data which is older than {}", transactionIdToDelete);
     byte[] startKey = TransactionalKvBackendImpl.generateCommitKey(transactionIdToDelete);
-    byte[] endKey = endOfTransactionId();
+    commitIdHasBeenCollected = kvBackend.get(LAST_COLLECT_COMMIT_ID_KEY);
+    if (commitIdHasBeenCollected == null) {
+      commitIdHasBeenCollected = endOfTransactionId();
+    }
+
+    long lastGCId = getTransactionId(getBinaryTransactionId(commitIdHasBeenCollected));
+    LOG.info(
+        "Start to collect data which is modified between '{}({})' and '{}({})'",
```

Comment:
What is the definition of "between", is it `[a, b)` or `[a, b]`?

Reply:
Yeah, Assume the last GC has just collected data modified before time a (include the boundary), and now time is b and we are going to trigger another GC. In this time, we will only collect data that has been modified during the time a and b.

Reply:
So, what is the actual definition of "between"? You still don't explain anything.

Reply:
`Between` means a specific time range with a low boundary and high boundary , `between a and b` which means we would only collect the data that is modified in the time range [a, b], take an example, in first time, the current time is '2024-04-16 08:00:00', we will do full gc and collect data from minimum time 1, so the range is [1,  1713225600(2024-04-16 08:00:00)], the next trigger time is  '2024-04-16 09:00:00', then the range is [1713225600(2024-04-16 08:00:00), 1713229200(2024-04-16 08:00:00)].